web/packages/teleport/src/AppLauncher/AppLauncher.tsx

```diff
@@ -41,6 +41,21 @@ export function AppLauncher() {
     const port = location.port ? `:${location.port}` : '';
 
     try {
+      // Attempt to resolve the FQDN of the app, if we can't then an error
+      // will be returned preventing a redirect to a potentially arbitrary
+      // address.
+      //
+      // Ignore the public addr that was passed; if the public addr 
+      // matches the public app of an app then the user will get redirected
+      // to whatever params.fqdn is set to. So, only the params.fqdn is
+      // taken into account during the lookup.
+      await service.getAppFqdn({
+        fqdn: params.fqdn,
+        clusterId: params.clusterId,
+        publicAddr: params.fqdn,
```

Comment:
I can see a potential issue where a legitimate request to launch a valid app is denied due to the `fqdnHint` not being enough to resolve the app because I don't pass the `publicAddr` when doing an app lookup. If someone more familiar with app access thinks this could be a potential issue I have an idea on how to mitigate this while still disallowing redirection to unknown URLs.

Reply:
Is there a way for this check to happen on the backend? I'm a bit naive here.

cc @kimlisa , can you take a look at this when you get a chance? I know you were working around this area in the past.

Reply:
can we pass the params as is? and have the backend check if the fqdn passed in, and the resulting fqdn equals each other? or am i completely off the mark? 😅

Reply:
also i think @avatus is right. could we possibly check if fqdn is valid starting somewhere [here](https://github.com/gravitational/teleport/blob/dc1199636558f85ed275512175a12207a8034d44/lib/web/app/auth.go#L55)?

Reply:
> I can see a potential issue where a legitimate request to launch a valid app is denied due to the fqdnHint not being enough to resolve the app because I don't pass the publicAddr when doing an app lookup

tbh, i wasn't completely sure what this meant, can this scenario be tested?

Reply:
Per the slack conversation, this check can't happen in the backend, but by comparing the resolved FQDN to the passed FQDN this should be a non-issue.